spyder/plugins/editor/lsp/transport/common/consumer.py

```diff
@@ -126,3 +129,7 @@ def parse_headers(self, headers):
     def stop(self):
         with self.mutex:
             self.stopped = True
+
+    def read_num_bytes(self, n):
+        """Subclasses should override this method"""
+        return b''
```

Comment:
Perhaps this should be

    raise NotImplementedError

?

I mean, that should force subclassess to really implement this.

Reply:
You're right